pkg/server/server_test.go

```diff
@@ -0,0 +1,152 @@
+/*
+ *
+ * Copyright 2023. Metacontroller authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * /
+ */
+
+package server
+
+import (
+	"strings"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const errShouldBeNil = "err should be nil, got: %v"
+
+// see: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
+func TestTargetLabelSelectorParsing(t *testing.T) {
+	var err error
+	// Single
+	_, err = labels.Parse("foo=bar")
```

Comment:
we should not test `labels.Parse` method, I believe it is tested by upstream

Reply:
Thanks @grzesuav . I have removed the tests for this. I initially added them just to test my logic, but they don't test the code path. I will look to adding some e2e tests.